src/libertem/udf/stddev.py

```diff
@@ -0,0 +1,189 @@
+
+    src
+        A buffer that contains sum of variances, sum of frames, and the
+        number of frames used over current iteration of partition
+    """
+    p0 = VariancePart(var=dest['var'][:],
+                    sum_im=dest['sum_frame'][:],
+                    N=dest['num_frame'][:])
+    p1 = VariancePart(var=src['var'][:],
+                    sum_im=src['sum_frame'][:],
+                    N=src['num_frame'][:])
+    compute_merge = merge(p0, p1)
+
+    dest['var'][:] = compute_merge.var
+    dest['sum_frame'][:] = compute_merge.sum_im
+    dest['num_frame'][:] = compute_merge.N
+
+
+def merge(p0, p1):
+    """
+    Given two sets of partitions, with sum of frames
+    and sum of variances, compute joint sum of frames
+    and sum of variances using one pass algorithm
+
+    Parameters
+    ----------
+    p0
+        Contains information about the first partition, including
+        sum of variances, sum of pixels, and number of frames used
+
+    p1
+        Contains information about the second partition, including
+        sum of variances, sum of pixels, and number of frames used
+
+    Returns
+    -------
+    collections.namedtuple object
+        Contains information about the merged partitions, including
+        sum of variances, sum of pixels, and number of frames used
+    """
+    if p0.N == 0:
+        return VariancePart(var=p1.var, sum_im=p1.sum_im, N=p1.N)
+    N = p0.N + p1.N
+
+    # compute mean for each partitions
+    mean_A = (p0.sum_im / p0.N)
+    mean_B = (p1.sum_im / p1.N)
+
+    # compute mean for joint samples
+    delta = mean_B - mean_A
+    mean = mean_A + (p1.N * delta) / (p0.N + p1.N)
+
+    # compute sum of images for joint samples
+    sum_im_AB = p0.sum_im + p1.sum_im
+
+    # compute sum of variances for joint samples
+    delta_P = mean_B - mean
+    var_AB = p0.var + p1.var + (p1.N * delta * delta_P)
+
+    return VariancePart(var=var_AB, sum_im=sum_im_AB, N=N)
+
+
+def run_stddev(ctx, dataset):
+    """
+    Compute sum of variances and sum of pixels from the given dataset
+
+    Parameters
+    ----------
+    ctx
+        Context class that contains methods for loading datasets, creating jobs on them
+        and running them
+
+    dataset
+        dataset to work on
+
+    Returns
+    -------
+    pass_results
+        A buffer that contains sum of variances, sum of pixels, and
+        number of frames used to compute the above statistic
+
+    Example
+    -------
+    To retrieve statistic, using the following commands:
+    variance : pass_results['stddev']
+    standard deviation : pass_results['std']
+    sum of pixels : pass_results['sum_frame']
+    mean : pass_results['mean']
+    number of frames : pass_results['num_frame']
+
+
+    """
+    pass_results = ctx.run_udf(
+        dataset=dataset,
+        fn=compute_batch,
+        make_buffers=batch_buffer,
+        merge=batch_merge,
+    )
+
+    pass_results['var'] = pass_results['var'].data/pass_results['num_frame'].data
```

Comment:
Your return type is a bit inconsistent: for `var`, `std`, and `mean` you return `ndarray` instances; `num_frame` and `sum_frame`  are `BufferWrapper`s. You should decide on one here; I guess `ndarray` for all makes sense?

I discussed with @kruzaeva before that having a separate return type would be cleaner than returning `BufferWrapper`s; when we get there we can change the return type here.

Reply:
I've made all of them into into narray instances now